drivers/espi/espi_saf_mchp_xec.c

```diff
@@ -0,0 +1,906 @@
+	}
+
+ecp_exit:
+	k_sem_give(&xdat->ecp_lock);
+
+	return rc;
+}
+
+/* Flash read using SAF EC Portal */
+static int saf_xec_flash_read(const struct device *dev,
+			      struct espi_saf_packet *pckt)
+{
+	LOG_DBG("%s", __func__);
+	return saf_ecp_access(dev, pckt, MCHP_SAF_ECP_CMD_CTYPE_READ0);
+}
+
+/* Flash write using SAF EC Portal */
+static int saf_xec_flash_write(const struct device *dev,
+			       struct espi_saf_packet *pckt)
+{
+	return saf_ecp_access(dev, pckt, MCHP_SAF_ECP_CMD_CTYPE_WRITE0);
+}
+
+/* Flash erase using SAF EC Portal */
+static int saf_xec_flash_erase(const struct device *dev,
+			       struct espi_saf_packet *pckt)
+{
+	return saf_ecp_access(dev, pckt, MCHP_SAF_ECP_CMD_CTYPE_ERASE0);
+}
+
+
+static int espi_saf_xec_manage_callback(const struct device *dev,
+					struct espi_callback *callback,
+					bool set)
+{
+	struct espi_saf_xec_data *data = DEV_DATA(dev);
+
+	return espi_manage_callback(&data->callbacks, callback, set);
+}
+
+static int espi_saf_xec_activate(const struct device *dev)
+{
+	const struct espi_saf_xec_config *cfg;
+	MCHP_SAF_HW_REGS *regs;
+
+	if (dev == NULL) {
+		return -EINVAL;
+	}
+
+	cfg = DEV_CFG(dev);
+	regs = (MCHP_SAF_HW_REGS *)cfg->saf_base_addr;
+
+	regs->SAF_FL_CFG_MISC |= MCHP_SAF_FL_CFG_MISC_SAF_EN;
+
+	return 0;
+}
+
+static int espi_saf_xec_init(const struct device *dev);
+
+static const struct espi_saf_driver_api espi_saf_xec_driver_api = {
+	.config = espi_saf_xec_configuration,
+	.set_protection_regions = espi_saf_xec_set_pr,
+	.activate = espi_saf_xec_activate,
+	.get_channel_status = espi_saf_xec_channel_ready,
+	.flash_read = saf_xec_flash_read,
+	.flash_write = saf_xec_flash_write,
+	.flash_erase = saf_xec_flash_erase,
+	.manage_callback = espi_saf_xec_manage_callback,
+};
+
+static struct espi_saf_xec_data espi_saf_xec_data;
+
+static const struct espi_saf_xec_config espi_saf_xec_config = {
+	.saf_base_addr = DT_INST_REG_ADDR_BY_IDX(0, 0),
+	.qmspi_base_addr = DT_INST_REG_ADDR_BY_IDX(0, 1),
+	.saf_comm_base_addr = DT_INST_REG_ADDR_BY_IDX(0, 2),
+#if DT_INST_NODE_HAS_PROP(0, poll_timeout)
+	.poll_timeout = DT_INST_PROP(inst, poll_timeout),
+#else
+	.poll_timeout = MCHP_SAF_FLASH_POLL_TIMEOUT,
+#endif
+#if DT_INST_NODE_HAS_PROP(0, consec_rd_timeout)
+	.consec_rd_timeout = DT_INST_PROP(inst, consec_rd_timeout),
+#else
+	.consec_rd_timeout = MCHP_SAF_FLASH_CONSEC_READ_TIMEOUT,
+#endif
+#if DT_INST_NODE_HAS_PROP(0, sus_chk_delay)
+	.sus_chk_delay = DT_INST_PROP(inst, sus_chk_delay),
+#else
+	.sus_chk_delay = MCHP_SAF_FLASH_SUS_CHK_DELAY,
+#endif
+#if DT_INST_NODE_HAS_PROP(0, sus_rsm_interval)
+	.sus_rsm_interval = DT_INST_PROP(inst, sus_rsm_interval),
+#else
+	.sus_rsm_interval = MCHP_SAF_FLASH_SUS_RSM_INTERVAL,
+#endif
+#if DT_INST_NODE_HAS_PROP(0, poll_interval)
+	.poll_interval = DT_INST_PROP(inst, poll_interval),
+#else
+	.poll_interval = MCHP_SAF_FLASH_POLL_INTERVAL,
+#endif
+};
+
+DEVICE_AND_API_INIT(espi_saf_xec_0, DT_INST_LABEL(0),
```

Comment:
Use DEVICE_DT_INST_DEFINE instead.

Reply:
Hi. Thanks for the review.
I made the above changes, rebased to current master, and pushed it.